pydatastructs/strings/__init__.py

```diff
@@ -1,8 +1,18 @@
 __all__ = []
 
-from . import trie
+from . import (
+    trie,
+    string_matching_algorithms
```

Comment:
Please rename the file as `algorithms.py` from `string_matching_algorithms.py`. We would keep all the string related algorithms in this file.